vertx-core/src/main/java/io/vertx/core/impl/HostnameResolver.java

```diff
@@ -129,18 +120,65 @@ public Future<Void> close() {
     return provider.close();
   }
 
-  public static int parseNdotsOptionFromResolvConf(String s) {
+  // visible for testing
+  public static ResolverOptions parseLinux(File f) {
+    try {
+      if (f.exists() && f.isFile()) {
+        return parseLinux(new String(Files.readAllBytes(f.toPath())));
+      }
+    } catch (Throwable t) {
+      log.debug("Failed to load options from /etc/resolv.conf", t);
+    }
+    return new ResolverOptions(DEFAULT_NDOTS, DEFAULT_ROTATE);
+  }
+
+  // exists mainly to facilitate testing
+  public static ResolverOptions parseLinux(String input) {
     int ndots = -1;
-    Matcher matcher = NDOTS_OPTIONS_PATTERN.matcher(s);
-    while (matcher.find()) {
-      ndots = Integer.parseInt(matcher.group(1));
+    boolean rotate = false;
+    try {
+      int optionsIndex = input.indexOf(OPTIONS_ROW_LABEL);
+      if (optionsIndex != -1) {
+        boolean isProperOptionsLabel = false;
+        if (optionsIndex == 0) {
+          isProperOptionsLabel = true;
+        } else if (Character.isWhitespace(input.charAt(optionsIndex - 1))) {
+            isProperOptionsLabel = true;
+        }
+        if (isProperOptionsLabel) {
+          String afterOptions = input.substring(optionsIndex + OPTIONS_ROW_LABEL.length());
+          int rotateIndex = afterOptions.indexOf(ROTATE_LABEL);
+          if (rotateIndex != -1) {
+            if (!containsNewLine(afterOptions.substring(0, rotateIndex))) {
+              rotate = true;
+            }
+          }
+          int ndotsIndex = afterOptions.indexOf(NDOTS_LABEL);
+          if (ndotsIndex != -1) {
+            if (!containsNewLine(afterOptions.substring(0, ndotsIndex))) {
+              Matcher matcher = Pattern.compile("ndots:[ \\t\\f]*(\\d)+(?=$|\\s)").matcher(afterOptions.substring(ndotsIndex));
```

Comment:
I see you're not compiling this anymore.

I suppose you have good reasons (like it being used only once?).

Reply:
It should probably be moved into a holder instead

Reply:
Done